backend/portal/admin.py

```diff
@@ -1,19 +1,85 @@
 from django.contrib import admin
 from django.utils.html import escape, mark_safe
 
-from portal.models import Poll, PollOption, PollVote, Post, TargetPopulation
+from portal.models import Content, Poll, PollOption, PollVote, Post, TargetPopulation
 
 
-class PostAdmin(admin.ModelAdmin):
-    def image_tag(self, instance):
-        return mark_safe('<img src="%s" height="300" />' % escape(instance.image.url))
+class ContentAdmin(admin.ModelAdmin):
+    @admin.action(description="Set status to Approved")
+    def action_approved(modeladmin, request, queryset):
+        queryset.update(status=Content.STATUS_APPROVED)
 
-    image_tag.short_description = "Post Image"
-    readonly_fields = ("image_tag",)
+    @admin.action(description="Set status to Draft")
+    def action_draft(modeladmin, request, queryset):
+        queryset.update(status=Content.STATUS_DRAFT)
+
+    @admin.action(description="Set status to Revision")
+    def action_revision(modeladmin, request, queryset):
+        queryset.update(status=Content.STATUS_REVISION)
+
+    actions = [action_approved, action_draft, action_revision]
+
+    def get_queryset(self, request):
+        queryset = super().get_queryset(request)
+        return queryset.annotate(ar=Content.ACTION_REQUIRED_CONDITION).order_by(
+            "-ar", "created_date"
+        )
+
+    def ar(self, obj):
+        return obj.ar
+
+    ar.boolean = True
+
+
+class PostAdmin(ContentAdmin):
+    def image_tag(instance, height):
+        return mark_safe(
+            f'<img src="%s" height="{height}" />' % escape(instance.image and instance.image.url)
+        )
+
+    def small_image(self, instance):
+        return PostAdmin.image_tag(instance, 100)
+
+    small_image.short_description = "Post Image"
+
+    def large_image(self, instance):
+        return PostAdmin.image_tag(instance, 300)
+
+    large_image.short_description = "Post Image"
+
+    readonly_fields = ("large_image",)
+    list_display = (
+        "title",
+        "subtitle",
+        "club_code",
+        "club_comment",
+        "start_date",
+        "expire_date",
+        "ar",
+        "status",
+        "small_image",
+    )
+
+
+class PollOptionInline(admin.TabularInline):
+    model = PollOption
+    extra = 0
+
+
+class PollAdmin(ContentAdmin):
+    inlines = [PollOptionInline]
+    list_display = (
+        "question",
+        "club_code",
+        "club_comment",
+        "start_date",
+        "expire_date",
+        "ar",
+        "status",
+    )
 
 
 admin.site.register(TargetPopulation)
-admin.site.register(Poll)
-admin.site.register(PollOption)
+admin.site.register(Poll, PollAdmin)
 admin.site.register(PollVote)
```

Comment:
Any reason you didn't inline `PollVote` as well?

Reply:
PollVote is embedded inline into the poll admin view. idk didn't seem nescecary to have stand alone so it would've just been more clutter

Reply:
Sounds good